consensus/ising/proposerservice.go

```diff
@@ -755,15 +762,27 @@ func (ps *ProposerService) SetOrChangeMind(votingType voting.VotingContentType,
 }
 
 func (ps *ProposerService) HandleProposalMsg(proposal *Proposal, sender *crypto.PubKey) {
-	// TODO: vote counting for cached blocks
+	hash := *proposal.hash
+	height := proposal.height
+	nodeID := publickKeyToNodeID(sender)
+
+	// handle proposal when block syncing
 	if ps.localNode.GetSyncState() != protocol.PersistFinished {
+		err := ps.syncCache.AddVoteForBlock(hash, height, nodeID)
+		if err != nil {
+			return
+		}
+		block, err := ps.syncCache.GetBlockFromSyncCache(height)
+		if err != nil {
+			return
+		}
+		ps.localNode.SetSyncStopHash(block.Header.PrevBlockHash, block.Header.Height-1)
```

Comment:
One potential security concern for this approach is that malicious node can continuously send wrong block to nodes in sync mode to prevent it from finish syncing

Reply:
There's a flag in SetSyncStopHash function when setting stop hash. If there's a valid stop hash set, the later will not be set again.

Reply:
The first one is voted by neighbors. If most neighbors are evil, let it go.